internal/generate/clusterserviceversion/clusterserviceversion.go

```diff
@@ -179,32 +179,35 @@ func makeCSVFileName(name string) string {
 }
 
 // makeKustomizeBaseGetter returns a function that gets a kustomize-style base.
-func (g Generator) makeKustomizeBaseGetter(inputDir, apisDir string) getBaseFunc {
+func (g Generator) makeKustomizeBaseGetter(inputDir, apisDir string, ilvl projutil.InteractiveLevel) getBaseFunc {
 	basePath := filepath.Join(inputDir, "bases", makeCSVFileName(g.OperatorName))
 	if genutil.IsNotExist(basePath) {
 		basePath = ""
 	}
 
-	return g.makeBaseGetter(basePath, apisDir)
+	return g.makeBaseGetter(basePath, apisDir, ilvl)
 }
 
 // makeBaseGetter returns a function that gets a base from inputDir.
 // apisDir is used by getBaseFunc to populate base fields.
-func (g Generator) makeBaseGetter(basePath, apisDir string) getBaseFunc {
+func (g Generator) makeBaseGetter(basePath, apisDir string, ilvl projutil.InteractiveLevel) getBaseFunc {
 	gvks := make([]schema.GroupVersionKind, len(g.config.Resources))
 	for i, gvk := range g.config.Resources {
 		gvks[i].Group = fmt.Sprintf("%s.%s", gvk.Group, g.config.Domain)
 		gvks[i].Version = gvk.Version
 		gvks[i].Kind = gvk.Kind
 	}
 
+	baseExists := genutil.IsNotExist(basePath)
 	return func() (*operatorsv1alpha1.ClusterServiceVersion, error) {
 		b := bases.ClusterServiceVersion{
 			OperatorName: g.OperatorName,
 			OperatorType: g.OperatorType,
 			BasePath:     basePath,
 			APIsDir:      apisDir,
 			GVKs:         gvks,
+			// Check if user explicitly wants an interactive prompt or has no preference.
+			Interactive: (ilvl == projutil.InteractiveSoftOff && baseExists) || ilvl == projutil.InteractiveOnAll,
```

Comment:
As discussed before.
Might be worth covering this with a unit test if possible.
```suggestion
			Interactive: (ilvl == projutil.InteractiveSoftOff && baseNotExists) || ilvl == projutil.InteractiveOnAll,
```

Reply:
Done.